crates/esthri/src/ops/sync.rs

```diff
@@ -153,6 +153,58 @@ where
     Ok(())
 }
 
+/// Syncs between S3 prefixes and local directories
+///
+/// # Arguments
+///
+/// * `s3` - S3 client
+/// * `source` - S3 prefix or local directory to sync from
+/// * `destination` - local directory to sync to
+/// * `glob_filter` - An (optional) slice of filters that specify whether files
+///                   should be included or not. These are processed in order,
+///                   with the first matching filter determining whether the
+///                   file will be included or excluded. If not supplied, then
+///                   all files will be synced.
+#[logfn(err = "ERROR")]
+pub async fn sync_down_streaming<'a, T>(
+    s3: &'a T,
+    source: &'a S3PathParam,
+    destination: &'a S3PathParam,
+    filters: &'a [GlobFilter],
+    compressed: bool,
+) -> Result<impl Stream<Item = Result<Option<String>>> + 'a>
```

Comment:
I'm wondering if instead of `Item = Result<Option<String>>` maybe something like `Item = Result<SyncResult>`

```rust
pub enum SyncResult  {
    Synced(String),
    Cached(String),
}
impl SyncResult {
    pub fn into_path(self) -> String { ... }
}
```

because for my particular use case I still need the names of the already downloaded files. I suppose I could just walk the directory before syncing and chain that stream with the one returned here. Could go either way. Also don't really like the name/variant names but can't think of anything better

Reply:
ended up creating a struct, wdyt? @notoriaga 
```
pub enum Synced {
    /// newly created file after sync
    Created,
    /// already existed file
    Existed,
}

pub struct SyncedFile {
    synced: Synced,
    src_path: String,
    dest_path: String,
}
```